r2dbc-spi/src/main/java/io/r2dbc/spi/CredentialFactory.java

```diff
@@ -0,0 +1,16 @@
+package io.r2dbc.spi;
+
+/** Factory for creating {@link Credential} objects */
+public class CredentialFactory {
+
```

Comment:
I'd suggest adding a private constructor that does nothing:
```java
private CredentialFactory() { }
```
Typically, we don't want to instantiate factories, as their methods are all static.

You may want to declare the class itself as final. This indicates that the class is not designed for inheritance.

Reply:
agreed, resolved with commit 9ce228d